ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,185 @@
+        model: 'database/role',
+        path: 'vault.cluster.secrets.backend.credentials',
+      };
+    case 'issue-certificate-pki':
+      return {
+        title: 'Role to use',
+        elementType: 'search-select',
+        placeholder: 'Type to find a role...',
+        buttonText: 'Issue leaf certificate',
+        model: 'pki/role',
+        path: 'vault.cluster.secrets.backend.pki.roles.role.generate',
+      };
+    case 'view-certificate-pki':
+      return {
+        title: 'Certificate serial number',
+        placeholder: '33:a3:...',
+        elementType: 'search-select',
+        buttonText: 'View certificate',
+        model: 'pki/certificate/base',
+        path: 'vault.cluster.secrets.backend.pki.certificates.certificate.details',
+      };
+    case 'view-issuer-pki':
+      return {
+        title: 'Issuer',
+        placeholder: 'Type issuer name or ID',
+        elementType: 'search-select',
+        buttonText: 'View issuer',
+        model: 'pki/issuer',
+        nameKey: 'issuerName',
+        path: 'vault.cluster.secrets.backend.pki.issuers.issuer.details',
+      };
+  }
+};
+
+const PKI_ENGINE_TYPE = 'pki';
+const KV_ENGINE_TYPE = 'kv';
+const DB_ENGINE_TYPE = 'database';
+export default class DashboardQuickActionsCard extends Component {
+  @service router;
+
+  @tracked selectedEngine;
+  @tracked selectedActions = [];
+  @tracked selectedAction;
+  @tracked actionParamField;
+  @tracked selectedEngineName;
+  @tracked value;
+
+  constructor() {
+    super(...arguments);
+
+    if (!this.selectedEngine) {
+      this.setSelectedAction();
+    }
+  }
+
+  get filteredSecretEngines() {
+    return this.args.secretsEngines.filter(
+      (secretEngine) =>
+        secretEngine.type === PKI_ENGINE_TYPE ||
+        secretEngine.type === KV_ENGINE_TYPE ||
+        secretEngine.type === DB_ENGINE_TYPE
+    );
+  }
+
+  get secretsEnginesOptions() {
+    return this.filteredSecretEngines.map((engine) => {
+      const { id, type } = engine;
+      return { name: id, type, id };
+    });
+  }
+
+  @action
+  handleSearchEngineSelect([selection]) {
+    if (selection) {
+      this.selectedEngine = selection.type;
+      this.selectedEngineName = selection.id;
+      this.selectedActions = getActionsByEngineType(this.selectedEngine);
+      this.setSelectedAction(this.selectedActions?.firstObject.actionType);
+    } else {
+      // no selection, clear tracked properties
+      this.selectedEngine = '';
+      this.selectedEngineName = '';
+      this.setSelectedAction();
+    }
+  }
+
+  @action
+  setSelectedAction(selectedAction) {
+    if (selectedAction) {
+      this.selectedAction = selectedAction;
+      this.actionParamField = getActionParamByAction(selectedAction);
+    } else {
+      this.selectedActions = getActionsByEngineType('kv');
+      this.actionParamField = getActionParamByAction('find-kv');
+    }
+  }
+
+  @action
+  handleSearch(val) {
+    if (Array.isArray(val)) {
+      this.value = val[0];
```

Comment:
Same request here - `value` made it a little hard to intuit what this is used for perhaps `actionParamValue`?